### DISCUSSION TITLE
Using The tailwind autocomplete and intellisense in other language modes

### DISCUSSION THREAD

**Discussion by JMinyard1335** [tailwindlabs/tailwindcss#14254] [Help]
So Ive looked around on this topic and I have seen some talk about using the the vs code extension with other languages. My thing is im trying to get it to work with the org mode extension. In org mode you can write a markdown like language and export it to html. since my primary editor is emacs I use org mode a lot and would love to integrate it more into my vscode work flow. currently my setting look like: 
![image](https://github.com/user-attachments/assets/723f2916-6aed-41ad-b757-5598580a3c28)
while my tailwind.config.js looks like: 
![image](https://github.com/user-attachments/assets/f6383785-4015-4cda-ab0f-3eab6e9cf045)
and the idea is to get the intellisense and auto complete to work with this type of syntax:
![image](https://github.com/user-attachments/assets/1c5291ed-aebf-45a1-acd4-b2c681ef9c1b)
Ik this might seem like a stupid idea but personally I would rather look at an org document then raw html any day of the week and getting this set up makes it easier to move projects back and forth between the editors. any advice would be appreciated or if what im trying is not possible lmk. 

Side note heres whats happening with my set up. It seems that tailwind will search for my classes in the org doc fine as when the org doc has no tailwind classes and i run the tailwind cli command to watch and build the css file it tells me no classes in the src folder but when i put the classes back in the document it does not complain. Secondly If i use the other org formatting such as source blocks  then it seems to work:
![image](https://github.com/user-attachments/assets/edb418a8-33eb-4514-99b4-64beda2db718)

**Comment by wongjn** [comment]
Consider checking you have the correct language ID for the `.org` files. `Org` looks wrong to me due to the uppercase character, though I'm not sure if `org` is the language ID either – I'd double check it matches exactly.

Futhermore, your regex in `tailwindCss.experimental.classRegex` (and by extension, your extractor function) seems incorrect. You may have meant to escape the `+` character, since it has special meaning in regular expressions. Also, the capture group syntax seems incorrect. You have a hat character `^` inside the one-of construct `[]` but then it seems you tried to close the capture group without supplying any characters inside the one-of construct.

```diff
 "tailwindCss.experimental.classRegex": [
-  "#+ATTR_HTML: :class ([^]*)"
+  "#\\+ATTR_HTML: :class (.*)"
 ]
```
---
As an aside, please consider not [providing your code as images](https://meta.stackoverflow.com/questions/285551). It makes it harder for us to copy the code out and experiment with it. As images, other users would need to retype it out. This takes more effort that some may not wish to undertake, compared to copying code as text.

**Reply by JMinyard1335** [reply]
You my friend are a life savior. It has been awhile since i messed with regex you were correct. And I appreciate the aside I will do that in the future. Thanks again happy coding!
